SQL/New_patches/2023-02-17_add_unique_indices_to_mri_violation_tables.sql

```diff
@@ -0,0 +1,12 @@
+-- ---------------------------------------------------------------------------------------------
+-- alter MRICandidateErrors table to add PhaseEncodingDirection and EchoNumber
+-- ---------------------------------------------------------------------------------------------
+ALTER TABLE MRICandidateErrors ADD COLUMN `PhaseEncodingDirection` VARCHAR(3)  DEFAULT NULL;
+ALTER TABLE MRICandidateErrors ADD COLUMN `EchoNumber`             VARCHAR(20) DEFAULT NULL;
+
+
+ALTER TABLE MRICandidateErrors ADD UNIQUE KEY `unique_MCE_entry` (`SeriesUID`, `EchoTime`, `PhaseEncodingDirection`, `EchoNumber`, `PatientName`, `Reason`);
+
+ALTER TABLE mri_violations_log ADD UNIQUE KEY `unique_mvl_entry` (`SeriesUID`, `EchoTime`, `PhaseEncodingDirection`, `EchoNumber`, `Scan_type`, `Severity`, `Header`, `Value`, `ValidRange`, `ValidRegex`);
```

Comment:
@driusan @ridz1208 @laemtl when running this statement on my VM, I get the following error:

```
ERROR 1071 (42000): Specified key was too long; max key length is 3072 bytes
```
Not sure how to resolve this...

Reply:
funny thing: it worked on my mac. Glad I had the idea to test it on my VM as well!

Reply:
FYI for the context:
- `SeriesUID`, `EchoTime`, `PhaseEncodingDirection`, `EchoNumber` determines a unique NIfTI or MINC file
- `Scan_type`, `Severity`, `Header`, `Value`, `ValidRange`, `ValidRegex` determines a unique violation for the NIfTI/MINC file

Reply:
@nicolasbrossard suggested to implement this https://planet.mysql.com/entry/?id=5990991/ for the `mri_violations_log` unique index. @driusan @ridz1208 @laemtl can you let me know your thoughts?

I know the PR is failing the tests right now, it is very likely RB needs to be updated but before spending lots of time fixing RB, I'd like a confirmation that what I suggest make sense to you guys. Thanks!

Reply:
Hash values can collide, would it be an issue since it can prevent a legit tupple to be inserted?